plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineProjectDeployer.java

```diff
@@ -0,0 +1,63 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import java.io.File;
+import java.util.Collections;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+
+import com.google.cloud.tools.appengine.api.deploy.DefaultDeployConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDeployment;
+import com.google.common.annotations.VisibleForTesting;
+
+public class AppEngineProjectDeployer {
+
+  private static final String PLUGIN_ID = "com.google.cloud.tools.eclipse.appengine.localserver"; //$NON-NLS-1$
+  private static final String APPENGINE_WEB_XML_PATH = "WEB-INF/appengine-web.xml"; //$NON-NLS-1$
+
+  private AppEngineDeployInfo deployInfo;
+
+  public AppEngineProjectDeployer() {
+    deployInfo = new AppEngineDeployInfo();
+  }
+
+  @VisibleForTesting
+  AppEngineProjectDeployer(AppEngineDeployInfo deployInfo) {
+    this.deployInfo = deployInfo;
+  }
+
+  public void deploy(IPath stagingDirectory, CloudSdk cloudSdk, IProgressMonitor monitor) throws CoreException {
```

Comment:
It would seem to me that this should be more a utility method that parses & obtains the `AppEngineDeployInfo` and then calls to a `deploy(AppEngineDeployInfo info, Path staging, etc)`.  That would allow an override of the `AppEngineDeployInfo` information.